README.md

```diff
@@ -1,17 +1,106 @@
-# Griptape Tool Template
+# Griptape Plugins Template
 
-A Github template repository for creating Griptape tools.
+A Github template repository for creating Griptape plugins.
 
-## Usage
+## Getting Started
 
-This template repository defines the typical structure for a Griptape tool and is meant to be used as a starting point when developing new tools. To create a tool using this template, click the **"Use this template"** button at the top of the repository.
+Via `gh`:
 
-### Steps for creating a new Griptape tool:
+```
+$ gh repo create griptape-plugin-name -p griptape/griptape-plugin-template
+```
+
+Via github web page:
+
+Click on `Use this template`
+
+![](https://docs.github.com/assets/cb-36544/images/help/repository/use-this-template-button.png)
+
+## What is a Griptape Plugin?
+
+A Griptape Plugin is nothing more than a python class that implements one of Griptape's interfaces.
+
+This repository provides a recommended structure for organizing your plugin code, as well as helpful tools for testing and development.
+
+## Plugin Structure
+
+The template repository is structured as follows:
+
+├── griptape
+│   └── plugin_name # Name whatever you want
+│       └── tools
+│           └── reverse_string
+│               └── tool.py
+            ...more directories for other interfaces (drivers, tasks, structures, etc)...
+└── tests
+    └── unit
+        └── tools
+            └── test_reverse_string_tool.py
+├── examples
+│   └── example.py # Example usage of the plugin 
+├── LICENSE # Choose the appropriate license
+├── Makefile # Contains useful commands for development
+├── pyproject.toml # Contains the project's metadata
+├── README.md #  Describes the plugin and how to use it
+
+## Development
+
+### One Command Setup
+
+Run the following command to do a one-time setup of the project:
+
+```bash
+make setup
+```
+
+You can also run the commands individually:
+
+### Installing Dependencies
+
+```bash
+make install
+```
+
+### Running Tests
+
+```bash
+make test
+```
+
+### Running Checks
+
+```bash
+make check
+```
+
+### Running Formatter
+
+```bash
+make format
+```
+
+### Running Example
+
+1. Set appropriate environment variables. The example requires an `OPENAI_API_KEY` environment variable to be set.
```

Comment:
can you do a `1a.` that says "each example will require its own environment variables. See the XXX in each example to understand which environment variables are required to run them."